Build Fails Due to Tests

MVN
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
Maven home: /usr/local/Cellar/maven/3.6.3_1/libexec
Java version: 11.0.9.1, vendor: AdoptOpenJDK, runtime: /Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.16", arch: "x86_64", family: "mac"

Results:

Errors:
  GradleProjectImporterTest.testGradleJavaHome:231 ArrayIndexOutOfBounds Index 0...
  GradleProjectImporterTest.testJavaHome:207 ArrayIndexOutOfBounds Index 0 out o...

Tests run: 1288, Failures: 0, Errors: 2, Skipped: 12

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for JDT Language Server :: Parent 0.68.0-SNAPSHOT:
[INFO]
[INFO] JDT Language Server :: Parent ...................... SUCCESS [  0.118 s]
[INFO] JDT Language Server :: Target Platform ............. SUCCESS [  0.203 s]
[INFO] JDT Language Server :: Core ........................ SUCCESS [ 21.734 s]
[INFO] JDT Language Server :: Tests ....................... FAILURE [07:33 min]
[INFO] JDT Language Server :: SyntaxServer Tests .......... SKIPPED
[INFO] JDT Language Server :: Product ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  08:28 min
[INFO] Finished at: 2021-01-18T10:15:46-08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:2.0.0:test (default-test) on project org.eclipse.jdt.ls.tests: There are test failures.
[ERROR]
[ERROR] Please refer to /Users/anshumankomawar/eclipse.jdt.ls/org.eclipse.jdt.ls.tests/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :org.eclipse.jdt.ls.tests

Indeed I can! Same error:

pmr@shadow org.eclipse.jdt.ls.tests % ../mvnw clean verify -X
...
Results:

Errors: 
  GradleProjectImporterTest.testGradleJavaHome:228 ArrayIndexOutOfBounds Index 0...
  GradleProjectImporterTest.testJavaHome:204 ArrayIndexOutOfBounds Index 0 out o...

Tests run: 1281, Failures: 0, Errors: 2, Skipped: 12

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  05:46 min
[INFO] Finished at: 2021-02-24T00:39:19-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:2.0.0:test (default-test) on project org.eclipse.jdt.ls.tests: There are test failures.
[ERROR] 
[ERROR] Please refer to /Users/pmr/IdeaProjects/eclipse.jdt.ls/org.eclipse.jdt.ls.tests/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:2.0.0:test (default-test) on project org.eclipse.jdt.ls.tests: There are test failures.

Please refer to /Users/pmr/IdeaProjects/eclipse.jdt.ls/org.eclipse.jdt.ls.tests/target/surefire-reports for the individual test results.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
Caused by: org.apache.maven.plugin.MojoFailureException: There are test failures.

Please refer to /Users/pmr/IdeaProjects/eclipse.jdt.ls/org.eclipse.jdt.ls.tests/target/surefire-reports for the individual test results.
    at org.eclipse.tycho.surefire.TestMojo.runTest (TestMojo.java:1063)
    at org.eclipse.tycho.surefire.TestMojo.execute (TestMojo.java:712)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.apache.maven.wrapper.BootstrapMainStarter.start (BootstrapMainStarter.java:39)
    at org.apache.maven.wrapper.WrapperExecutor.execute (WrapperExecutor.java:122)
    at org.apache.maven.wrapper.MavenWrapperMain.main (MavenWrapperMain.java:55)
[ERROR] 
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
pmr@shadow org.eclipse.jdt.ls.tests % java -version
openjdk version "11.0.9.1" 2020-11-04
OpenJDK Runtime Environment AdoptOpenJDK (build 11.0.9.1+1)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 11.0.9.1+1, mixed mode)
pmr@shadow org.eclipse.jdt.ls.tests %

@snjeza Still fails for me both before and after removing JAVA_HOME and GRADLE_HOME


Results:

Errors: 
  GradleProjectImporterTest.testGradleJavaHome:237 ArrayIndexOutOfBounds Index 0...
  GradleProjectImporterTest.testJavaHome:209 ArrayIndexOutOfBounds Index 0 out o...

Tests run: 1284, Failures: 0, Errors: 2, Skipped: 12

[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for JDT Language Server :: Parent 0.70.0-SNAPSHOT:
[INFO] 
[INFO] JDT Language Server :: Parent ...................... SUCCESS [  0.044 s]
[INFO] JDT Language Server :: Target Platform ............. SUCCESS [  0.089 s]
[INFO] JDT Language Server :: Core ........................ SUCCESS [ 16.414 s]
[INFO] JDT Language Server :: Tests ....................... FAILURE [05:26 min]
[INFO] JDT Language Server :: SyntaxServer Tests .......... SKIPPED
[INFO] JDT Language Server :: Product ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  05:50 min
[INFO] Finished at: 2021-03-03T23:00:35-05:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.eclipse.tycho:tycho-surefire-plugin:2.0.0:test (default-test) on project org.eclipse.jdt.ls.tests: There are test failures.
[ERROR]

--- a/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java
+++ b/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JVMConfigurator.java
@@ -56,6 +56,8 @@ import org.eclipse.lsp4j.MessageType;
  */
 public class JVMConfigurator implements IVMInstallChangedListener {
 
+	public static final String MAC_OSX_VM_TYPE = "org.eclipse.jdt.internal.launching.macosx.MacOSXType"; //$NON-NLS-1$
+
 	public static boolean configureDefaultVM(String javaHome) throws CoreException {
 		if (StringUtils.isBlank(javaHome)) {
 			return false;
@@ -74,6 +76,10 @@ public class JVMConfigurator implements IVMInstallChangedListener {
 		IVMInstall vm = findVM(jvmHome, null);
 		if (vm == null) {
 			IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+			if (installType == null || installType.getVMInstalls().length == 0) {
+				// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
+				installType = JavaRuntime.getVMInstallType(MAC_OSX_VM_TYPE);
+			}
 			long unique = System.currentTimeMillis();
 			while (installType.findVMInstall(String.valueOf(unique)) != null) {
 				unique++;
@@ -108,6 +114,10 @@ public class JVMConfigurator implements IVMInstallChangedListener {
 					IPath sourcePath = runtime.getSourcePath();
 					IVMInstall vm = findVM(file, runtime.getName());
 					IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+					if (installType == null || installType.getVMInstalls().length == 0) {
+						// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
+						installType = JavaRuntime.getVMInstallType(MAC_OSX_VM_TYPE);
+					}
 					VMStandin vmStandin;
 					if (vm == null) {
 						long unique = System.currentTimeMillis();
@@ -133,7 +143,7 @@ public class JVMConfigurator implements IVMInstallChangedListener {
 						continue;
 					}
 
-					
+
 					vmStandin.setName(runtime.getName());
 					vmStandin.setInstallLocation(file);
 
@@ -209,7 +219,7 @@ public class JVMConfigurator implements IVMInstallChangedListener {
 			connection.sendActionableNotification(runtimeNotification);
 			return;
 		}
-		
+
 		connection.showNotificationMessage(MessageType.Error, message);
 	}
 

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporterTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/GradleProjectImporterTest.java
@@ -51,6 +51,7 @@ import org.eclipse.jdt.internal.launching.StandardVMType;
 import org.eclipse.jdt.launching.IVMInstall;
 import org.eclipse.jdt.launching.IVMInstallType;
 import org.eclipse.jdt.launching.JavaRuntime;
+import org.eclipse.jdt.ls.core.internal.JVMConfigurator;
 import org.eclipse.jdt.ls.core.internal.JavaLanguageServerPlugin;
 import org.eclipse.jdt.ls.core.internal.JobHelpers;
 import org.eclipse.jdt.ls.core.internal.ProjectUtils;
@@ -200,6 +201,10 @@ public class GradleProjectImporterTest extends AbstractGradleBasedTest{
 		IVMInstall defaultVM = JavaRuntime.getDefaultVMInstall();
 		try {
 			IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+			if (installType == null || installType.getVMInstalls().length == 0) {
+				// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
+				installType = JavaRuntime.getVMInstallType(JVMConfigurator.MAC_OSX_VM_TYPE);
+			}
 			IVMInstall[] vms = installType.getVMInstalls();
 			IVMInstall vm = vms[0];
 			JavaRuntime.setDefaultVMInstall(vm, new NullProgressMonitor());
@@ -224,6 +229,10 @@ public class GradleProjectImporterTest extends AbstractGradleBasedTest{
 		IVMInstall defaultVM = JavaRuntime.getDefaultVMInstall();
 		try {
 			IVMInstallType installType = JavaRuntime.getVMInstallType(StandardVMType.ID_STANDARD_VM_TYPE);
+			if (installType == null || installType.getVMInstalls().length == 0) {
+				// https://github.com/eclipse/eclipse.jdt.ls/issues/1646
+				installType = JavaRuntime.getVMInstallType(JVMConfigurator.MAC_OSX_VM_TYPE);
+			}
 			IVMInstall[] vms = installType.getVMInstalls();
 			IVMInstall vm = vms[0];
 			JavaRuntime.setDefaultVMInstall(vm, new NullProgressMonitor());

